### PR TITLE
add unittest && change parseConfig interface

### PR DESCRIPTION
**Pull request by sulenn**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/go_client.go  11
- .ci/ethPrecompiled/bn256_gm.go  1
- conf/config_test.go  7
- client/api_handler.go  6
- precompiled/cns/cns_info.go  2
- conf/parse_pem.go  9
- .ci/ethPrecompiled/bn256.go  1
- conf/parse_pem_test.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5855895&bid=19170950)

File: .ci/ethPrecompiled/bn256.go [hunk 1]
```diff
@@ -282,7 +283,10 @@ func (_BN256 *BN256TransactorSession) Bn256ScalarMul(x [32]byte, y [32]byte, sca
 }
 
 func main() {
-	configs := conf.ParseConfig("config.toml")
+	configs, err := conf.ParseConfig("config.toml")
```
Comment (bxq2011hust) [inline .ci/ethPrecompiled/bn256.go:286]:
```suggestion
	configs, err := conf.ParseConfigFile("config.toml")
```

File: client/api_handler.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func (api *APIHandler) SendRawTransaction(ctx context.Context, groupID int, tx *
 		fmt.Printf("rlp encode tx error!")
 		return err
 	}
-	return api.CallContext(ctx, nil, "sendRawTransaction", groupID, common.ToHex(data))
+	return api.CallContext(ctx, nil, "sendRawTransaction", groupID, hexutil.Encode(data))
```
Comment (bxq2011hust) [inline client/api_handler.go:100]: good job, std library is perfered

File: conf/config.go [hunk 3]
```diff
@@ -23,7 +22,7 @@ type Config struct {
 }
 
 // ParseConfig parses the configuration from toml config file
-func ParseConfig(cfgFile string) []Config {
+func ParseConfig(cfgFile string) ([]Config, error) {
```
Comment (bxq2011hust) [inline conf/config.go:25]:
```suggestion
func ParseConfigFile(cfgFile string) ([]Config, error) {
```

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- .ci/ethPrecompiled/bn256.go  1
- precompiled/cns/cns_info.go  2
- conf/parse_pem_test.go  5
- conf/config_test.go  4
- conf/parse_pem.go  9
- client/api_handler.go  6
- .ci/ethPrecompiled/bn256_gm.go  1
- client/go_client.go  11
         

Complexity decreasing per file
==============================
+ conf/config.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5855895&bid=19170950)

File: conf/parse_pem_test.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package conf
+
+import (
+	"testing"
+)
+
+const (
+	standardKeyHex      = "b89d42f12290070f235fb8fb61dcf96e3b11516c5d4f6333f26e49bb955f8b62"
+	standardCurve       = "secp256k1"
```
Comment (bxq2011hust) [inline conf/parse_pem_test.go:9]: add sm2p256v1 also

File: conf/test.toml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+[Network]
+#type rpc or channel
+Type="channel"
```
Comment (bxq2011hust) [inline conf/test.toml:3]: delete this file

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- precompiled/cns/cns_info.go  2
- conf/config_test.go  4
- conf/parse_pem.go  9
- conf/parse_pem_test.go  9
- client/api_handler.go  6
- client/go_client.go  11
- .ci/ethPrecompiled/bn256_gm.go  1
- .ci/ethPrecompiled/bn256.go  1
         

Complexity decreasing per file
==============================
+ conf/config.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5855895&bid=19170950)

**Pull request merged**